src/key/helper.js

```diff
@@ -355,35 +355,70 @@ export function sanitizeKeyOptions(options, subkeyDefaults = {}) {
   return options;
 }
 
-export function isValidSigningKeyPacket(keyPacket, signature) {
-  const keyAlgo = keyPacket.algorithm;
-  return keyAlgo !== enums.publicKey.rsaEncrypt &&
-    keyAlgo !== enums.publicKey.elgamal &&
-    keyAlgo !== enums.publicKey.ecdh &&
-    (!signature.keyFlags ||
-      (signature.keyFlags[0] & enums.keyFlags.signData) !== 0);
+export function validateSigningKeyPacket(keyPacket, signature, config) {
+  switch (keyPacket.algorithm) {
+    case enums.publicKey.rsaEncryptSign:
+    case enums.publicKey.rsaSign:
+    case enums.publicKey.dsa:
+    case enums.publicKey.ecdsa:
+    case enums.publicKey.ed25519Legacy:
+    case enums.publicKey.ed25519: {
+      if (!signature.keyFlags && !config.allowMissingKeyFlags) {
+        throw new Error('None of the key flags is set: consider passing `config.allowMissingKeyFlags`');
+      }
+
+      return !signature.keyFlags ||
+        (signature.keyFlags[0] & enums.keyFlags.signData) !== 0;
+    }
```

Comment:
Sorry for missing these last time (and being very nitpicky ^.^), but these brackets are also not necessary. They're only necessary if you need a block scope, e.g. if you're creating block-scoped variables.
```suggestion
    case enums.publicKey.ed25519:
      if (!signature.keyFlags && !config.allowMissingKeyFlags) {
        throw new Error('None of the key flags is set: consider passing `config.allowMissingKeyFlags`');
      }

      return !signature.keyFlags ||
        (signature.keyFlags[0] & enums.keyFlags.signData) !== 0;
```

Reply:
I know they are not needed, but it's not a big deal to add them...? I just find it confusing to have a lot of code without brackets.

Reply:
Er, OK, but I think it's more important to have a consistent code style. JS cases normally don't have brackets, and most other code in the library doesn't have them, so we should only add them when needed.

Reply:
Ok, but then I'll add a linting rule for this, so that it's automatically enforced